lib/Jsrt/Jsrt.cpp

```diff
@@ -4644,6 +4644,13 @@ CHAKRA_API JsCreateWeakReference(
             return JsErrorNoCurrentContext;
         }
 
+        if (Js::TaggedNumber::Is(value)) {
```

Comment:
I was considering making this check be `!Js::RecyclableObject::Is(value)` instead, but I wasn't sure how to validate the incoming weak reference in `JsGetWeakReferenceValue`. Are weak reference pointers guaranteed to have any particular shape?

Reply:
Reading the code, looks weak reference pointer is the same normal pointer as other pointers. Why do you consider that alternative? It looks to me `TaggedNumber::Is` works fine and probably better. Used elsewhere as well.

nit: Consider moving this fragment up out of GlobalAPIWrapper. It is simple operation and no need to enter the wrapper.

Reply:
I was considering checking for `RecyclableObject` since that would be potentially more future-proof in case we add non-number non-recyclable objects.

I'll move the code out of the wrapper as you suggest.